core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -111,6 +111,49 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     zkConsumerConnector1.shutdown()
   }
 
+  @Test
+  def testBrokerTopicMetrics() {
+    val replicationBytesIn = "ReplicationBytesInPerSec"
+    val replicationBytesOut = "ReplicationBytesOutPerSec"
+    val bytesIn = "BytesInPerSec,topic=" + topic
+    val bytesOut = "BytesOutPerSec,topic=" + topic
+
+    createTopic(zkUtils, topic, 1, numNodes, servers)
+    // Topic metrics can take a moment to be created
+    TestUtils.waitUntilTrue(
+        () => Metrics.defaultRegistry.allMetrics.asScala.filterKeys(k => k.getMBeanName.endsWith(topic)).size > 1,
```

Comment:
Hmm, the BytesIn metric is only created when there is data produced to that topic. So, perhaps we should publish some messages first?